library/src/domain/Co2Eq/Co2Eq.ts

```diff
@@ -0,0 +1,111 @@
+import * as Either from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as Codec from "io-ts/lib/Codec";
+import * as Decoder from "io-ts/lib/Decoder";
+import { Co2EqUnit, Co2EqUnitEnum } from "./Co2EqUnit";
+
+export class Co2Eq {
+  averageInGrams;
+
+  constructor({ averageValue, unit }: Co2EqProps) {
+    this.averageInGrams = convertToGrams(averageValue, unit);
+  }
+
+  format(co2EqUnit?: Co2EqUnitEnum) {
+    const unit = co2EqUnit
+      ? co2EqUnit
+      : findBestUnitForAmountInGrams(this.averageInGrams);
+
+    switch (unit) {
+      case Co2EqUnitEnum.KILOGRAM:
+        return `${this.averageInGrams / 1e3} kg`;
+
+      case Co2EqUnitEnum.TON:
+        return `${this.averageInGrams / 1e6} tons`;
+
+      case Co2EqUnitEnum.GRAM:
+        return `${this.averageInGrams} g`;
+    }
+  }
+
+  static build(co2EqProps: Co2EqProps) {
+    return new Co2Eq(co2EqProps);
+  }
+
+  static factory({
+    averageValue = 100 * 1000,
+    unit = Co2EqUnitEnum.KILOGRAM,
+  }: Partial<Co2EqProps> = {}) {
+    return Co2Eq.build({ averageValue, unit });
+  }
+
+  static propsCodec = Codec.struct({
+    averageValue: Codec.number,
+    unit: Co2EqUnit.codec,
+  });
+
+  static stringCodec = Codec.make<unknown, string, Co2Eq>(
+    pipe(
+      Decoder.string,
+      Decoder.compose({
+        decode: (string) => {
+          const REGEX = /^([0-9]*,?\.?[0-9]*) *(g|kg|ton)$/;
```

Comment:
Since below you replace `,` by `.`:

```suggestion
          const REGEX = /^([0-9]*\.?[0-9]*) *(g|kg|ton)$/;
```